extensions/true-fantom/network/network-v001.js

```diff
@@ -0,0 +1,477 @@
+                value: '3'
+              },
+              {
+                text: '(4) status',
+                value: '4'
+              },
+              {
+                text: '(5) status text',
+                value: '5'
+              },
+              {
+                text: '(6) type',
+                value: '6'
+              },
+              {
+                text: '(7) redirected?',
+                value: '7'
+              },
+              {
+                text: '(8) url',
+                value: '8'
+              }
+            ]
+          },
+          response_type: {  
+            acceptReporters: true,
+            items: [
+              {
+                text: '(1) text',
+                value: '1'
+              },
+              {
+                text: '(2) json',
+                value: '2'
+              },
+              {
+                text: '(3) status ok?',
+                value: '3'
+              },
+              {
+                text: '(4) status',
+                value: '4'
+              },
+              {
+                text: '(5) status text',
+                value: '5'
+              },
+              {
+                text: '(6) type',
+                value: '6'
+              },
+              {
+                text: '(7) redirected?',
+                value: '7'
+              },
+              {
+                text: '(8) url',
+                value: '8'
+              },
+              {
+                text: '(9) body used?',
+                value: '9'
+              }
+            ]
+          },
+          default: { 
+            acceptReporters: true,
+            items: [
+              {
+                text: 'default',
+                value: 'default'
+              }
+            ]
+          }
+        }
+      }
+    }
+    
+    connected_to_internet_block() {
+      return navigator.onLine;
+    }
+    browser_block() {
+      let has = input => navigator.userAgent.includes(input);
+      if (has('Firefox')) return 'Firefox';
+      if (has('SamsungBrowser')) return 'Samsung Internet';
+      if ((has('Opera') || has('OPR')) && has('GX')) return 'Opera GX';
+      if (has('Opera') || has('OPR')) return 'Opera';
+      if (has('Trident')) return 'Internet Explorer';
+      if (has('Edge')) return 'Edge Legacy';
+      if (has('Edg')) return 'Edge Chromium';
+      if (has('YaBrowser') || has('YaSearchBrowser')) return 'Yandex';
+      if (has('Miui')) return 'Mi Browser';
+      if (has('UBrowser')) return 'UC Browser';
+      if (has('Chrome')) return 'Chromium';
+      if (has('Safari')) return 'Safari';
+      return '';
+    }
+    current_url_block() {
+      return document.URL || '';
+    }
+    network_type_block() {
+      let type = navigator.connection.type;
+      return type === undefined || type === 'none' || type === 'unknown' ? '' : type;
+    }
+    network_generation_block() {
+      return navigator.connection.effectiveType || '';
```

Comment:
These blocks need to gracefully fallback when someone using a browser that does not have `navigator.connection` such as Firefox

Reply:
Yeah. This was in the version before "optimization"
I thought that `return something || ''` would work as `try catch`. I apologize for not checking this change.

And judging by the compatibility table, `try catch` will also be necessary for `navigator.onLine` block.

Ps: Can errors occur with `navigator.userAgent` `document.URL` `window.open` and `fetch()`?

Reply:
You may want to keep in mind that:

 - new URL() throws if the URL is invalid
 - fetch() returns a rejected promise in several conditions. you should attempt to handle this.

Reply:
navigator.onLine and document.URL are supported basically everywhere

 - https://caniuse.com/mdn-api_navigator_online
 - https://caniuse.com/mdn-api_document_url